packages/core/src/logging/DefaultLogger.ts

```diff
@@ -60,15 +60,27 @@ export class DefaultLogger implements Logger {
     this.debugMode = debugMode;
   }
 
-  isEnabled(namespace: LoggerNamespace): boolean {
+  isEnabled(namespace: LoggerNamespace, context?: LogContext) {
+    if (context?.isDisabled !== undefined && context.isDisabled !== false) {
+      if (context.isDisabled === true) {
+        // Disable for ALL
+        return false;
+      } else if (context.isDisabled === 'on-success'
+        && context.level
+        && !['warning', 'error'].includes(context.level)) {
+        // Disable only on success and the level was NOT an error state
+        return false;
+      }
```

Comment:
one code style thing + no need for `else if` when you `return`

```suggestion
      }

      if (
        context.isDisabled === 'on-success' &&
        context.level &&
        !['warning', 'error'].includes(context.level)
      ) {
        // Disable only on success and the level was NOT an error state
        return false;
      }
```

btw i am not completely sold on the idea of the `on-success` value. if a query fails, it will result in error which has the query inside the error message already, i think we dont need to enforce it here when logging is disabled. try it in a real app to see, iirc it works this way.

that's also the reason why i was asking for some integration tests, to see how it will actually work for users

Reply:
Oh sorry, I didn't think you'd re-review after I resolved the merge stuff... I had some of that stuff locally but just wanted to fix the branch after the rebase. here's what I'm planning:

<img width="637" alt="image" src="https://user-images.githubusercontent.com/6445731/234653062-a4353454-768e-48a9-bc43-5b658c03ef97.png">

`enabled` will act as a master override; if `disabled` no logs will be output. Period.

If `debugMode` is set, it'll be used instead of what the logger was initialized with as a local override.

Sound good?

Reply:
for me this is about the find methods mainly, i'd keep it, you need to be able to set what to log (query or query + query params), right?

i guess the flush method should support this too somehow

Reply:
Just pushed the changes; let me know what you'd like to do. My recommendation is to pull the `debugMode` for now and add it back later when it's actually relevant

Reply:
Yea eventually I'd love for this to be added to `update`, `flush`, and `persistAndFlush`... that was something I hoped to work on next. If you're good with it as is, I can update the docs just not totally sure how 😬

Reply:
Before v6 branch gets merged to master, those docs are not deployed anywhere, so we can have something in progress there, not a huge deal.

I'd like to merge it within a few weeks, let's say around half of may.